[management] Implement Job API with Validation, Persistence, and Bundle Job Support

aliamerj opened the pull request:
## Describe your changes

Implemented the API for job operations, including creating new jobs, fetching jobs by peer, and fetching individual job details. Added support for `bundle/debug` jobs with fully validated parameters (e.g., `bundle_for_time`, `log_file_count`, `anonymize`) and ensured that pending jobs are handled correctly.

Temporary permissions are based on the existing “delete peers” permission to allow job operations, with a TODO to implement a dedicated permission for creating and fetching jobs. Also added TODOs to integrate with the job channel once it is ready  #4367

All job data is persisted in the database, including account ID, peer ID, job type, status, parameters, and results. Jobs are validated before creation to prevent invalid payloads.

## Issue ticket number and link #4354

## Stack

- `feature/remote-debug` - :warning: No PR associated with branch <!-- branch-stack -->
  - \#4384 :point\_left:

### Checklist

- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation

Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (explain why)

### Docs PR URL (required if "docs added" is checked)

Paste the PR link from <https://github.com/netbirdio/docs> here:

<https://github.com/netbirdio/docs/pull/>\_\_

pascal-fischer commented on management/server/store/sql_store.go:
```diff
@@ -124,6 +126,75 @@ func GetKeyQueryCondition(s *SqlStore) string {
+		}
+	}()
+
+	return s.db.WithContext(ctx).
```
please remove the WithContext from the db methods. We had some bad experience with this about calls beeing canceld due to cotext although the API (or grpc)  request was still ongoing

pascal-fischer commented on management/server/store/sql_store.go:
```diff
@@ -124,6 +126,75 @@ func GetKeyQueryCondition(s *SqlStore) string {
+	}()
+
+	return s.db.WithContext(ctx).
+		Clauses(clause.OnConflict{UpdateAll: true}).
+		Create(job).Error
```
Instead of `SaveJob` use `CreatePeerJob` and `UpdatePeerJob` this allowes for better control and no accidental overwrites. Then you do not need a onConflict clause

pascal-fischer commented on management/server/activity/codes.go:
```diff
@@ -286,6 +286,8 @@ var activityMap = map[Activity]Code{
 
 	PeerIPUpdated: {"Peer IP updated", "peer.ip.update"},
+
+	JobCreatedByUser: {"Create Job for peer", "peer.job.created"},
```
Sorry, my bad. It's present, not past. The created is without the 'd'

aliamerj commented:
When using oneOf in `openapi.yml`, the `oapi-codegen/pkg/runtime` package must be installed to be work

"github.com/deepmap/oapi-codegen/pkg/runtime"

aliamerj commented:
I’ve updated the way we handle workloads. Instead of using JSON blobs or `interface{}/any`, I switched to full embedding with typed fields. This keeps everything strongly typed, avoids runtime casting, and is much cleaner to work with.
And it's stored in the database like this.

id | created_at | completed_at | triggered_by | peer_id | account_id | status | failed_reason | workload_workload_type | workload_anonymize | workload_bundle_for | workload_bundle_for_time | workload_log_file_count | workload_upload_key
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
b52427ac.. | 2025-08-29 07:14:39.515923738+00:00 |   | google-oauth2.. | d2fqqjj57uqqe46uchhg | d19eo0b57uqqji4hco20 | pending |   | bundle | 0 | 1 | 2 | 50 |  

aliamerj commented on management/server/types/job.go:
```diff
@@ -0,0 +1,148 @@
+type Workload struct {
+	Type JobType `gorm:"column:workload_type;index;type:varchar(50)"`
+
+	// Only one branch is valid depending on Type. The other branch stays zero-valued.
```
I used embedding so every field becomes its own column (flat table).
